src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java

```diff
@@ -610,6 +617,16 @@ public Context getContext() {
     }
 
     @Override
+    public byte getLuminanceFromSky(int x, int y, int z) {
+        return (byte) Math.max(0, getLightFor(EnumSkyBlock.SKY, new BlockPos(x, y, z)) - getSkylightSubtracted());
```

Comment:
NMS lacks a sever side accessible function to fetch the light level, with the time of day/weather factored in, thus this must be subtracted, and a floor is imposed to prevent negative light levels.